pkg/apis/pingcap/v1alpha1/types.go

```diff
@@ -1437,3 +1441,272 @@ type IngressSpec struct {
 	// +optional
 	TLS []extensionsv1beta1.IngressTLS `json:"tls,omitempty"`
 }
+
+// +k8s:openapi-gen=true
+// DMCluster is the control script's spec
+type DMCluster struct {
+	metav1.TypeMeta `json:",inline"`
+	// +k8s:openapi-gen=false
+	metav1.ObjectMeta `json:"metadata"`
+
+	// Spec defines the behavior of a dm cluster
+	Spec DMClusterSpec `json:"spec"`
+
+	// +k8s:openapi-gen=false
+	// Most recently observed status of the dm cluster
+	Status DMClusterStatus `json:"status"`
+}
+
+// +k8s:openapi-gen=true
+// DMDiscoverySpec contains details of Discovery members for dm
+type DMDiscoverySpec struct {
+	corev1.ResourceRequirements `json:",inline"`
+	// Host indicates the existed TiDB discovery host. If not specified, dm cluster will start a new one
+	Host string `json:"host"`
+}
+
+// +k8s:openapi-gen=true
+// DMClusterSpec describes the attributes that a user creates on a dm cluster
+type DMClusterSpec struct {
+	// Discovery spec
+	Discovery DMDiscoverySpec `json:"discovery,omitempty"`
+
+	// dm-master cluster spec
+	// +optional
+	Master *MasterSpec `json:"dm_master,omitempty"`
```

Comment:
The json tag should be `master`?

Reply:
`tiup-dm` uses `dm_master`
https://github.com/pingcap/tiup/blob/31ebd2a7bcaf32c1e0ebc87991818ab6011e83c2/components/dm/spec/topology_dm.go#L82

Reply:
@july2993 Can we change it `master_servers/worker_servers`?

Reply:
I think just `dm_master` is more concise?
there are no servers, just specified the master or worker's config.

like tiup-cluster.

Reply:
currently
```
server_configs:
    dm_master:
      rpc-timeout: "30s" # you many need update test_upgrade.sh if change this cause it's used to test edit-config
      rpc-rate-limit: 10.0
      rpc-rate-burst: 40
dm_master_servers:
    - host: 172.19.0.101
```

if change
```
server_configs:
    master_servers:
      rpc-timeout: "30s" # you many need update test_upgrade.sh if change this cause it's used to test edit-config
      rpc-rate-limit: 10.0
      rpc-rate-burst: 40
dm_master_servers:
    - host: 172.19.0.101
```

Reply:
`master` +1

Reply:
@july2993 Prefer
```yaml
server_configs:
    master:
      rpc-timeout: "30s" # you many need update test_upgrade.sh if change this cause it's used to test edit-config
      rpc-rate-limit: 10.0
      rpc-rate-burst: 40
master_servers:
    - host: 172.19.0.101
```